src/main/java/org/jenkinsci/plugins/ghprb/extensions/status/GhprbSimpleStatus.java

```diff
@@ -165,6 +174,11 @@ public void onBuildStart(AbstractBuild<?, ?> build,
         } else {
             sb.append(Ghprb.replaceMacros(build, listener, startedStatus));
         }
+
+        if(showMatrixStatus && !(build.getProject() instanceof MatrixProject)){
```

Comment:
Also with the logic you recommended, it will break the expected behaviour for matrix jobs. A matrix job that does want the status of its children job won't work. The checkbox will be false by default, and the parent job will return back true when checking if it's an instance of a matrix project and will thus exit before creating a commit status, when it should be creating a commit status.